patches/server/1027-Fix-MC-255756-shield-break-block-sounds-not-playing.patch

```diff
@@ -0,0 +1,66 @@
+From: roro1506HD <roro1506mcpro@gmail.com>
+Date: Fri, 25 Aug 2023 14:42:54 +0200
+Subject: [PATCH] Fix MC-255756 shield break/block sounds not playing
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+index e11d7283662834047b2ff81a2fd25a4263792deb..7e02de5ae22abeaad6607c5856e8326538716acb 100644
+--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
++++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
+@@ -1482,7 +1482,7 @@ public abstract class LivingEntity extends Entity implements Attackable {
+             boolean knockbackCancelled = this.level().paperConfig().environment.disableExplosionKnockback && source.is(DamageTypeTags.IS_EXPLOSION) && this instanceof net.minecraft.world.entity.player.Player; // Paper - Disable explosion knockback
+             if (flag1) {
+                 if (flag) {
+-                    this.level().broadcastEntityEvent(this, (byte) 29);
++                    this.level().playSound(null, this, SoundEvents.SHIELD_BLOCK, SoundSource.PLAYERS, 0.8F, 0.8F + this.level().random.nextFloat() * 0.4F); // Paper - Fixes MC-255756: Play the sound instead of the entity event
+                 } else {
+                     if (!knockbackCancelled) // Paper - Disable explosion knockback
+                     this.level().broadcastDamageEvent(this, source);
+@@ -2225,7 +2225,10 @@ public abstract class LivingEntity extends Entity implements Attackable {
+ 
+             // Apply blocking code // PAIL: steal from above
+             if (event.getDamage(DamageModifier.BLOCKING) < 0) {
+-                this.level().broadcastEntityEvent(this, (byte) 29); // SPIGOT-4635 - shield damage sound
++                // Paper start - Fixes MC-255756: Play the sound instead of the entity event
++                net.minecraft.world.entity.player.Player exemptedPlayer = this instanceof net.minecraft.world.entity.player.Player player ? player : null;
++                this.level().playSound(exemptedPlayer, this, SoundEvents.SHIELD_BLOCK, SoundSource.PLAYERS, 0.8F, 0.8F + this.level().random.nextFloat() * 0.4F);
++                // Paper end
+                 this.hurtCurrentlyUsedShield((float) -event.getDamage(DamageModifier.BLOCKING));
+                 Entity entity = damagesource.getDirectEntity();
+ 
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index e2a25c29ec74147b3e66aa0b3deb85a8f6ee53a5..f9a274932049e10e0a73b94e7e681cd37c0aaaa2 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -26,6 +26,8 @@ import net.minecraft.resources.ResourceLocation;
+ import net.minecraft.server.level.ServerLevel;
+ import net.minecraft.server.level.ServerPlayer;
+ import net.minecraft.sounds.SoundEvent;
++import net.minecraft.sounds.SoundEvents;
++import net.minecraft.sounds.SoundSource;
```

Comment:
Don't import - use fully qualified names instead.

Reply:
Oh yeah my bad, forgot to check the imports for this file